src/test/clojure/clara/test_durability.clj

```diff
@@ -254,4 +254,24 @@
             (check-fact expected-fact fact)))))))
  
 (deftest test-durability-fressian-serde
-  (durability-test :fressian))
+  (testing "SerDe of the rulebase along with working memory"
+    (durability-test :fressian))
+
+  (testing "Repeated SerDe of rulebase"
+    (let [rb-serde (fn [s]
+                     (with-open [baos (java.io.ByteArrayOutputStream.)]
+                       (d/serialize-rulebase s (df/create-session-serializer baos))
+                       (let [rb-data (.toByteArray baos)]
+                         (with-open [bais (java.io.ByteArrayInputStream. rb-data)]
+                           (d/deserialize-rulebase (df/create-session-serializer bais))))))
+
+          s (mk-session 'clara.durability-rules)
+          rb (-> s eng/components :rulebase)
+          deserialized1 (rb-serde s)
+          ;; Need a session to do the 2nd round of SerDe.
+          restored (d/assemble-restored-session deserialized1 {})
+          deserialized2 (rb-serde restored)]
+      
+      (is (= (-> rb :id-to-node keys set)
+             (-> deserialized1 :id-to-node keys set)
+             (-> deserialized2 :id-to-node keys set))))))
```

Comment:
Perhaps you could factor out the parts of durability-test that are just normal rule operations and assertions from the serialization parts and assert on the former here too?  This doesn't seem to do much to validate functional correctness after the second pass through serialization rather than just validating that the second deserialization doesn't cause an exception.  Functional correctness depends on the node structure itself; I don't even think the id-to-node map is used in the engine.